### DISCUSSION TITLE
Custom folder names for layers?

### DISCUSSION THREAD

**Discussion by shi-rudo** [feature-sliced/steiger#72] [Q&A]
Is there a possibility, e.g. in the `steiger.config.js` file, to specify custom names for the layers?

for example `src/app` --> `src/application`

**Comment by illright** [comment]
Currently not. Why do you want this feature? Feature-Sliced Design does not allow changing the layers' names. In this way, the benefit of consistency between projects is achieved

**Reply by daniilsapa** [reply]
@illright 
I recently thought about a possible case where such a feature might be useful. So, just providing this point to think about...

What if we try to apply FSD to mobile app development? I don't have much experience in Android or iOS development (in fact, my knowledge of Android development is Mariana-Trench-deep — limited to a university task where I made a calculator app 😂). But FSD potentially seems to fit good this domain too. Then, for example, the capability to rename "pages" to "views" would help people be closer to the lexicon they use to describe things in their domain (I highly doubt that Swift and Java support will ever be implemented in Steiger, but it could support React Native, Ionic, ... other JS-based mobile development approaches and maybe Flutter)

**Reply by illright** [reply]
Good point, we might consider that in the future

**Reply by Ivan-Gav** [reply]
> Currently not. Why do you want this feature? Feature-Sliced Design does not allow changing the layers' names. In this way, the benefit of consistency between projects is achieved

In Next.js you just can't use foldernames **app** and **pages** as FSD implies. They are reserved by the framework.

**Reply by illright** [reply]
As linked in another thread here, would this approach work for the time being?

https://feature-sliced.github.io/documentation/docs/guides/tech/with-nextjs#app-router

**Reply by Ivan-Gav** [reply]
Thank you for the feedback. But i think it has it's drawbacks too - it'll cause "fsd/insignificant-slice" for the pages imported into the next app folder outside src. Let alone moving part of the code outside src doesn't look as an elegant solution. I'd prefer custom layer names really.

**Comment by shi-rudo** [comment]
For example, if I use nextJS, I don't want to pollute the app router folder. As far as I know, nextJS does not allow this to be renamed either.

**Reply by illright** [reply]
Would the solution described here work for you?

https://feature-sliced.github.io/documentation/docs/guides/tech/with-nextjs#app-router

**Reply by TheOneAndOnlyGarcia** [reply]
Thanks for the suggestion, that should work.

**Reply by shi-rudo** [reply]
> Would the solution described here work for you?
> 
> https://feature-sliced.design/docs/guides/tech/with-nextjs#app-router

Good thing, fortunately, NextJS offers this option. Since our project is still quite young, that would be a viable option. Because not too many changes have to be made to our tooling workflow. Perhaps the suggestion for a custom mapping, as with the Feature-Sliced Design Helper intelliJ plugin, would be something for the future?
